projects/RabbitMQ.Client/client/impl/SocketFrameHandler.cs

```diff
@@ -101,21 +101,38 @@ class SocketFrameHandler : IFrameHandler
             _channelReader = channel.Reader;
             _channelWriter = channel.Writer;
 
-            if (ShouldTryIPv6(endpoint))
+            // Resolve the hostname to know if it's even possible to even try IPv6
+            IPAddress[] adds = Dns.GetHostAddresses(endpoint.HostName);
```

Comment:
This is a great opportunity to introduce a resolver interface similar to [that in the Java client](https://github.com/rabbitmq/rabbitmq-java-client/blob/master/src/main/java/com/rabbitmq/client/AddressResolver.java). This would allow the user to enforce, say, IPv6 connectivity.

Reply:
This goes beyond my capabilities.

Reply:
Fair enough.